sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -581,49 +593,56 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
         return new GlobalThroughputControlConfigBuilder(this, databaseId, containerId);
     }
 
-    void openConnectionsAndInitCaches() {
-        blockListVoidResponse(openConnectionsAndInitCachesInternal());
-    }
-
     WriteRetryPolicy getNonIdempotentWriteRetryPolicy() {
         return this.nonIdempotentWriteRetryPolicy;
     }
 
-    private Mono<List<Void>> openConnectionsAndInitCachesInternal() {
-        int concurrency = 1;
-        int prefetch = 1;
-        if (this.proactiveContainerInitConfig != null) {
-            return Flux.fromIterable(this.proactiveContainerInitConfig.getCosmosContainerIdentities())
-                    .flatMap(
-                        cosmosContainerIdentity -> Mono.just(this
-                            .getDatabase(containerIdentityAccessor.getDatabaseName(cosmosContainerIdentity))
-                            .getContainer(containerIdentityAccessor.getContainerName(cosmosContainerIdentity))),
-                        concurrency,
-                        prefetch
-                    )
-                    .flatMap(
-                        cosmosAsyncContainer -> cosmosAsyncContainer
-                            .openConnectionsAndInitCaches(
-                                this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount()),
-                        concurrency,
-                        prefetch)
-                    .collectList();
-        }
-        return Mono.empty();
+    void openConnectionsAndInitCaches() {
+        final Duration lastSuccessResponseTimeout = Duration.ofSeconds(5);
+
+        asyncDocumentClient.submitOpenConnectionTasksAndInitCaches(
```

Comment:
should the timeout be applied to the whole process? (getting the addresses + openConnections)?